### DISCUSSION TITLE
关于输入法多种语言输出需求

### DISCUSSION THREAD

**Discussion by ahoujanai** [rime/home#759] [Q&A]
**希望有一个快捷键能够在一秒钟之内切换简体，繁体，藏文，英文，俄文等，希望能够持续火力不间断的输出不同语言，或者能更快的在两种文字之间快速切换，比如简体和繁体，藏文跟俄文等。**

**Comment by lotem** [comment]
从[这里](https://github.com/biopolyhedron/rime_schemata)下载各语言的方案
然后分别设置[切换到方案](https://github.com/rime/rime-prelude/blob/master/key_bindings.yaml#L44)的快捷键

**Comment by ahoujanai** [comment]
> 从[这里](https://github.com/biopolyhedron/rime_schemata)下载各语言的方案
> 然后分别设置[切换到方案](https://github.com/rime/rime-prelude/blob/master/key_bindings.yaml#L44)的快捷键

虽然不太懂，但是还是谢谢你的回复

**Comment by toshiya14** [comment]
> 
> 
> > 从[这里](https://github.com/biopolyhedron/rime_schemata)下载各语言的方案
> > 然后分别设置[切换到方案](https://github.com/rime/rime-prelude/blob/master/key_bindings.yaml#L44)的快捷键
> 
> 虽然不太懂，但是还是谢谢你的回复

二楼回复得其实挺完整了，就是先安装你需要的输入方案/语言，然后各自设定一个快捷键，就可以了啊。

**Comment by shewer** [comment]
再問 可以設定 快鍵 切換 schema 嗎  目前 只知道ctrl-shift-1   可以 最近兩個方案交換 
不曉得如何 指定 方案 熱鍵   謝謝

**Comment by yu-soong** [comment]
@shewer 
在 default.yaml 里定义快捷键，如：

patch:
  key_binder/bindings:
    - { when: always, accept: Control+Shift+F2, select: luna_pinyin }
   
表示 Control+Shift+F2 切换到 luna_pinyin 这个 scheme 。
多说两句。为何要用这么一个复杂的快捷键，是因为我发现用别的某些快捷键会跟其他的程序有冲突。然后这个快捷方式就失效了。所以只好选了一个复杂的快捷键。
为方便起见，你可以用 AutoHotkey 等软件重新绑定 Control+Shift+F2 例如使单击 LShift  == Control+Shift+F2

**Reply by Immortal-Sty** [reply]
> @shewer 在 default.yaml 里定义快捷键，如：
> 
> patch: key_binder/bindings: - { when: always, accept: Control+Shift+F2, select: luna_pinyin }
> 
> 表示 Control+Shift+F2 切换到 luna_pinyin 这个 scheme 。 多说两句。为何要用这么一个复杂的快捷键，是因为我发现用别的某些快捷键会跟其他的程序有冲突。然后这个快捷方式就失效了。所以只好选了一个复杂的快捷键。 为方便起见，你可以用 AutoHotkey 等软件重新绑定 Control+Shift+F2 例如使单击 LShift == Control+Shift+F2

@yu-soong 

***

为啥我这里不行咧，我检查了快捷键没冲突，且换了很多其他快捷键，也都不行。

***

找到原因了，我把 `key_binder/bindings` 加上双引号就好了。

**Comment by librix** [comment]
我使用以下配置：中文可相互轉換，但外文不能轉換其他輸入方案，想要尋求幫助：
```javascript
# default.custom.yaml

patch:
  "key_binder/bindings/+":
    - { when: always, accept: Control+Shift+F1, select: wubi86_jidian }
    - { when: always, accept: Control+Shift+F2, select: double_pinyin }
    - { when: always, accept: Control+Shift+F3, select: cangjie5 }
    - { when: always, accept: Control+Shift+F4, select: cangjie6 }
    - { when: always, accept: Control+Shift+F5, select: kyril_international }
    - { when: always, accept: Control+Shift+F6, select: zhengma }
    - { when: always, accept: Control+Shift+F7, select: hebrew }
    - { when: always, accept: Control+Shift+F8, select: mipa }
    - { when: always, accept: Control+Shift+F9, select: qyeishanglr_hanja }
  schema_list:
    - {schema: luna_pinyin_fluency}
    - {schema: bopomofo_tw}
    - {schema: cangjie5}
    - {schema: double_pinyin}
    - {schema: wubi86_jidian}
    - {schema: cangjie6}
    - {schema: kyril_international}
    - {schema: zhengma}
    - {schema: hebrew}
    - {schema: mipa}
    - {schema: qyeishanglr_hanja}
```

**Reply by librix** [reply]
已经解决：转换失败的schema中定义 keybind：
```javascript
# qyeishanglr_hanja.custom.yaml
patch:
  "key_binder/bindings/+":
    - { when: always, accept: Control+Shift+F1, select: wubi86_jidian }
    - { when: always, accept: Control+Shift+F2, select: double_pinyin }
    - { when: always, accept: Control+Shift+F3, select: cangjie5 }
    - { when: always, accept: Control+Shift+F4, select: cangjie6 }
    - { when: always, accept: Control+Shift+F5, select: kyril_international }
    - { when: always, accept: Control+Shift+F6, select: zhengma }
    - { when: always, accept: Control+Shift+F7, select: hebrew }
    - { when: always, accept: Control+Shift+F8, select: mipa }
    - { when: always, accept: Control+Shift+F9, select: qyeishanglr_hanja }
```

**Comment by byzod** [comment]
我有个额外需求

安装中文和英文2个schema，并有快捷键切换，但是希望在不同窗口切换时，默认启用中文schema，就像是`switches`里的`reset: 0`效果一样，不过是针对schema，而不是states

具体来说，有以下3种输入模式：中文，中文+几个单词，（一段时间内）纯英文

混输模式的schema能解决前两个，而第3种则必须使用enter上屏，否则用空格就会出现`coming to`打成`coming 脱`，此时使用纯英文schema更好；但是因为主要是在打中文，如果没有默认模式，用过之后又要手动切回来，略麻烦（虽然也没有特别麻烦
